README.md

```diff
@@ -39,44 +43,165 @@ pip install git+https://github.com/arangoml/pyg-adapter.git
+    num_node_types=2,
+    num_edge_types=3,
+    avg_num_nodes=20,
+    avg_num_channels=3,  # avg number of features per node
+    edge_dim=2,  # number of features per edge
+    num_classes=3,  # number of unique label values
+)[0]
 
-# Let's assume that the ArangoDB "fraud detection" dataset is imported to this endpoint
+# Let's assume that the ArangoDB "IMDB" dataset is imported to this endpoint
 db = ArangoClient(hosts="http://localhost:8529").db("_system", username="root", password="")
 
-adbpyg_adapter = ADBPYG_Adapter(db, logging_lvl=logging.DEBUG)
+adbpyg_adapter = ADBPyG_Adapter(db)
+
+############################### PyG to ArangoDB ###############################
+
+# 1.1: PyG to ArangoDB
+adb_g = adbpyg_adapter.pyg_to_arangodb("FakeData", data)
+
+# 1.2: PyG to ArangoDB with a (completely optional) metagraph for customized adapter behaviour
+def y_tensor_to_2_column_dataframe(pyg_tensor):
+    label_map = {0: "Kiwi", 1: "Blueberry", 2: "Avocado"}
+
+    df = pandas.DataFrame(columns=["label_num", "label_str"])
+    df["label_num"] = pyg_tensor.tolist()
+    df["label_str"] = df["label_num"].map(label_map)
+
+    return df
+
+
+metagraph = {
+    "nodeTypes": {
+        "v0": {
+            "x": "features",  # 1) you can specify a string value for attribute renaming
+            "y": y_tensor_to_2_column_dataframe,  # 2) you can specify a function for user-defined handling, as long as the function returns a Pandas DataFrame
+        },
+    },
+    "edgeTypes": {
+        ("v0", "e0", "v0"): {
+            # 3) you can specify a list of strings for tensor dissasembly (if you know the number of node/edge features in advance)
+            "edge_attr": [ "a", "b"]  
+        },
+    },
+}
+
+
+adb_g = adbpyg_adapter.pyg_to_arangodb("FakeData", data, metagraph)
+
+# 1.3: PyG to ArangoDB with the same (optional) metagraph, but with `explicit_metagraph=True`
+# With `explicit_metagraph=True`, the node & edge types omitted from the metagraph will not be converted to ArangoDB.
+# Only 'v0' and ('v0', 'e0', 'v0') will be brought over (i.e 'v1', ('v0', 'e0', 'v1'), ... are ignored)
+adb_g = adbpyg_adapter.pyg_to_arangodb("FakeData", data, metagraph, explicit_metagraph=True)
+
+# 1.4: PyG to ArangoDB with a Custom Controller  (more user-defined behavior)
+class Custom_ADBPyG_Controller(ADBPyG_Controller):
+    def _prepare_pyg_node(self, pyg_node: dict, col: str) -> dict:
+        """Optionally modify a PyG node object before it gets inserted into its designated ArangoDB collection.
+
+        :param pyg_node: The PyG node object to (optionally) modify.
+        :param col: The ArangoDB collection the PyG node belongs to.
+        :return: The PyG Node object
+        """
+        pyg_node["foo"] = "bar"
+        return pyg_node
 
-# Use Case 1.1: ArangoDB to PyG via Graph name
-pyg_fraud_graph = adbpyg_adapter.arangodb_graph_to_pyg("fraud-detection")
+    # def _prepare_pyg_edge(self, pyg_edge: Json, edge_type: tuple) -> dict:
```

Comment:
This could either be removed or if this is meant to demonstrate that edges can also be modified, it might be worth adding a docstring to clarify?

Reply:
Yeah let's instead demonstrate that you can also modify edges, will address

Reply:
addressed in [ab082a5](https://github.com/arangoml/pyg-adapter/pull/2/commits/ab082a5921e78b1e9e6e0c679bb9f1f6210bf418)